www/pfSense-pkg-vHosts/files/usr/local/pkg/vhosts.inc

```diff
@@ -0,0 +1,367 @@
+
+        @custom_config
+
+        location / { index  index.html index.htm index.php; }
+
+        location ~ \.php$
+            {
+            try_files            \$uri =404; # This line closes a potential security hole
+                                             # ensuring users can't execute uploaded files
+                                             # see: http://forum.nginx.org/read.php?2,88845,page=3
+            fastcgi_pass         unix:/var/run/php-fpm.socket;
+            fastcgi_index        index.php;
+            fastcgi_param        SCRIPT_FILENAME  \$document_root\$fastcgi_script_name;
+            fastcgi_param        HTTP_PROXY  ""; # Fix httpoxy - https://httpoxy.org/#fix-now
+            fastcgi_read_timeout 180;
+            include              /usr/local/etc/nginx/fastcgi_params;
+            }
+
+        location = /favicon.ico  # Do not log requests for favicon.ico.
+            {
+            return               204;
+            access_log           off;
+            log_not_found        off;
+            }
+        }
+CONF_SERVER;
+
+/*-----------------------------------------*/
+/* nginx server SSL config block template. */
+/*                                         */
+/* @crt_file_path = Certificate file path. */
+/* @key_file_path = Private key file path. */
+/*-----------------------------------------*/
+$vhosts_g['conf_server_ssl'] = <<<CONF_SERVER_SSL
+
+        ssl_certificate                         '@crt_file_path';
+        ssl_certificate_key                     '@key_file_path';
+        ssl_session_timeout                     10m;
+        keepalive_timeout                       70;
+        ssl_session_cache                       shared:SSL:10m;
+        ssl_protocols                           TLSv1 TLSv1.1 TLSv1.2;
+        ssl_ciphers                             "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH";
+        ssl_prefer_server_ciphers               on;
+        add_header Strict-Transport-Security    "max-age=31536000";
+        add_header X-Content-Type-Options       nosniff;
+        ssl_session_tickets                     off;
+        ssl_dhparam                             /etc/dh-parameters.4096;
+CONF_SERVER_SSL;
+
+/*******************************************************************************
+* vhosts_build_service_config */
+/**
+* Rebuilds nginx configuration for vhosts. Called for package synchronization
+* and when any vhost is modified.
+*
+* Config file setting: <custom_php_resync_config_command>
+*******************************************************************************/
+function vhosts_build_service_config() {
+	global $config, $vhosts_g;
+
+	$conf_file_path   = $vhosts_g['conf_file_path'];
+	$conf_file_prefix = $vhosts_g['conf_file_prefix'];
+
+	/*-------------------------------------*/
+	/* Remove all existing configurations. */
+	/*-------------------------------------*/
+	conf_mount_rw();
+	exec("/bin/rm -f $conf_file_prefix*");
```

Comment:
Jim, from your lack of civility, I assumed you were a kid with no respect for those of us that have been here from the beginning. Looks like you made the same assumption about me. Following your last insults, I was pissed and stomped off like a child so nobody's behavior was sterling. But I know who you are, now. We're both long in the tooth in this business and would probably have a lot to talk about. You're wrong Jim. You're arguments are idiotic. Oh, I'll make this stupid change "full stop" because you're the gatekeeper and that appears matter to you. But I'd love it if you could give me a single example of how, without this "sanity check", there would be any danger to the router anywhere but on the test bench.